keyserver/listener.go

```diff
@@ -0,0 +1,106 @@
+package keyserver
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	. "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
+	defer ln.Close()
+
+	for {
+		select {
+		case <-server.stop:
+			server.waitCloseConn.Wait()
+			return
+		default:
+		}
+
+		conn, err := ln.Accept()
```

Comment:
What do you think about https://github.com/coniks-sys/coniks-go/blob/4ab5e299ac9f02d350c891cbf30a120362243fbd/keyserver/listener.go#L14-L30?